crates/ruff_python_semantic/src/definition.rs

```diff
@@ -187,7 +187,7 @@ impl<'a> Definitions<'a> {
     }
 
     /// Resolve the visibility of each definition in the collection.
-    pub fn resolve(self, exports: Option<&[&str]>) -> ContextualizedDefinitions<'a> {
+    pub fn resolve(self, exports: Option<&[DunderAllName]>) -> ContextualizedDefinitions<'a> {
```

Comment:
Nit: We could simplify this by changing `exports` to `&[DunderAllName]`. Call sites can pass an empty slice (or use `unwrap_or_default()` if there are no exports.

Reply:
Nice idea! Applied in https://github.com/astral-sh/ruff/pull/10525/commits/20e70c367a8a135989b55bc80e4f0cb4cb0734a6

Reply:
Hmm, that breaks things, and I think I know why: an empty export list (`__all__ = []`) is semantically different to a module not defining `__all__` at all. `__all__ = []` means there are no public re-exports from the module; not defining `__all__` at all means _everything_ is publicly re-exported from the module (unless it has a name starting with a single underscore).

Reply:
That makes sense, sorry. Thanks for trying.

Reply:
No worries!!